pkg/controller/apmserver/config.go

```diff
@@ -25,13 +29,43 @@ const (
 	CertificatesDir = "config/elasticsearch-certs"
 
 	APMServerHost        = "apm-server.host"
-	APMServerSecretToken = "apm-server.secret_token"
+	APMServerSecretToken = "apm-server.secret_token" // nolint
 
 	APMServerSSLEnabled     = "apm-server.ssl.enabled"
 	APMServerSSLKey         = "apm-server.ssl.key"
 	APMServerSSLCertificate = "apm-server.ssl.certificate"
+
+	ApmCfgSecretKey = "apm-server.yml" // nolint
 )
 
+// ReconcileConfig reconciles the configuration of the APM server: it first creates the configuration from the APM
+// specification and then reconcile the underlying secret.
+func ReconcileConfig(client k8s.Client, as *apmv1.ApmServer) (corev1.Secret, error) {
```

Comment:
We could make this function private like `reconcileApmServerToken` and `reconcileApmServerDeployment`.

Not sure but we could also rename it `reconcileApmServerConfig` for consistency and a little gain in readability against a little more verbosity?